python/cudf/tests/test_multiindex.py

```diff
@@ -0,0 +1,379 @@
+    assert_eq(pdf.loc[('a', 'store', 'storm', 'smoke')],
+              gdf.loc[('a', 'store', 'storm', 'smoke')])
+    # return 1 row and 1 remaining key = series
+    assert_eq(pdf.loc[('c', 'forest', 'clear')],
+              gdf.loc[('c', 'forest', 'clear')])
+    # assert_eq(pdf.loc[('a', 'store'): ('b', 'house')],
+    #           gdf.loc[('a', 'store'): ('b', 'house')])
+
+
+def test_multiindex_loc_then_column(pdf, gdf, pdfIndex):
+    gdfIndex = cudf.from_pandas(pdfIndex)
+    assert_eq(pdfIndex, gdfIndex)
+    pdf.index = pdfIndex
+    gdf.index = gdfIndex
+    assert_eq(pdf.loc[('a', 'store', 'clouds', 'fire')][0],
+              gdf.loc[('a', 'store', 'clouds', 'fire')][0])
+
+
+def test_multiindex_loc_rows_0(pdf, gdf, pdfIndex):
+    gdfIndex = cudf.from_pandas(pdfIndex)
+    pdf.index = pdfIndex
+    gdf.index = gdfIndex
+    with pytest.raises(KeyError):
+        print(pdf.loc[('d',)])
+    with pytest.raises(KeyError):
+        print(gdf.loc[('d',)])
+    assert_eq(pdf, gdf)
+
+
+def test_multiindex_loc_rows_1_2_key(pdf, gdf, pdfIndex):
+    gdfIndex = cudf.from_pandas(pdfIndex)
+    pdf.index = pdfIndex
+    gdf.index = gdfIndex
+    print(pdf.loc[('c', 'forest')])
+    print(gdf.loc[('c', 'forest')].to_pandas())
+    assert_eq(pdf.loc[('c', 'forest')], gdf.loc[('c', 'forest')])
+
+
+def test_multiindex_loc_rows_1_1_key(pdf, gdf, pdfIndex):
+    gdfIndex = cudf.from_pandas(pdfIndex)
+    pdf.index = pdfIndex
+    gdf.index = gdfIndex
+    print(pdf.loc[('c',)])
+    print(gdf.loc[('c',)].to_pandas())
+    assert_eq(pdf.loc[('c',)], gdf.loc[('c',)])
+
+
+def test_multiindex_column_shape():
+    pdf = pd.DataFrame(np.random.rand(5, 0))
+    gdf = cudf.from_pandas(pdf)
+    pdfIndex = pd.MultiIndex([['a', 'b', 'c']],
+                             [[0]])
+    pdfIndex.names = ['alpha']
+    gdfIndex = cudf.from_pandas(pdfIndex)
+    assert_eq(pdfIndex, gdfIndex)
+    with pytest.raises(ValueError):
+        pdf.columns = pdfIndex
+    with pytest.raises(ValueError):
+        gdf.columns = gdfIndex
+
+
+def test_multiindex_columns(pdf, gdf, pdfIndex):
+    pdf = pdf.T
+    gdf = cudf.from_pandas(pdf)
+    gdfIndex = cudf.from_pandas(pdfIndex)
+    assert_eq(pdfIndex, gdfIndex)
+    pdf.columns = pdfIndex
+    gdf.columns = gdfIndex
+    assert_eq(pdf[('a', 'store', 'clouds', 'fire')],
+              gdf[('a', 'store', 'clouds', 'fire')])
+    assert_eq(pdf[('a', 'store', 'storm', 'smoke')],
+              gdf[('a', 'store', 'storm', 'smoke')])
+    assert_eq(pdf[('a', 'store')],
+              gdf[('a', 'store')])
+    assert_eq(pdf[('b', 'house')],
+              gdf[('b', 'house')])
+    assert_eq(pdf[('a', 'store', 'storm')],
+              gdf[('a', 'store', 'storm')])
+    assert_eq(pdf[('a',)],
+              gdf[('a',)])
+    assert_eq(pdf[('c', 'forest', 'clear')],
+              gdf[('c', 'forest', 'clear')])
+    # assert_eq(pdf[('a', 'store'): ('b', 'house')],
+    #           gdf[('a', 'store'): ('b', 'house')])
```

Comment:
Stale code

Reply:
Reminders for a slice test that I intend to implement shortly into 0.8. I can create a new test to cover it and e-fail it, sound good?